apps/hash-ai-worker-ts/src/activities/infer-entities/propose-entities.ts

```diff
@@ -404,33 +424,47 @@ export const proposeEntities = async (params: {
             }
 
             inferenceState.proposedEntityCreationsByType = Object.entries(
-              proposedEntitiesByType,
+              validProposedEntitiesByType,
             ).reduce(
-              (prev, [entityTypeId, proposedEntitiesOfType]) => ({
+              (prev, [entityTypeId, proposedEntities]) => ({
                 ...prev,
                 [entityTypeId]: [
                   ...(prev[entityTypeId as VersionedUrl] ?? []),
-                  ...proposedEntitiesOfType.filter(
-                    ({ entityId }) =>
-                      // Don't include invalid entities
-                      !invalidProposedEntities.some(
-                        ({
-                          invalidProposedEntity: { entityId: invalidEntityId },
-                        }) => invalidEntityId === entityId,
-                      ) &&
-                      // Ignore entities we've inferred in a previous iteration, otherwise we'll get duplicates
-                      !inferenceState.proposedEntityCreationsByType[
-                        entityTypeId as VersionedUrl
-                      ]?.some(
-                        (existingEntity) =>
-                          existingEntity.entityId === entityId,
-                      ),
-                  ),
+                  ...proposedEntities,
                 ],
               }),
               inferenceState.proposedEntityCreationsByType,
             );
 
+            const now = new Date().toISOString();
+
+            if (validProposedEntities.length > 0) {
```

Comment:
In terms of what's shown in the UI, the logs are only used if the step outputs aren't populated yet... so as soon as the step outputs are available (i.e. the step is done), the contents of its outputs will replace whatever was taken from the logs in the table. 

I agree it would be best to have some kind of explicit 'proposal rejected / merged' log or something like that though